apps/studio/src/store/zustand/file.ts

```diff
@@ -0,0 +1,83 @@
+import { create } from "zustand";
+import { INITIAL_STATE } from "../getInitialState";
+
+export type File = {
+	hashDigest: string;
+	name: string;
+	content: string;
+	parent: string | null;
+};
+
+type FileState = {
+	files: Record<string, File>;
+	upsertOne: (hashDigest: string, newFile: File) => void;
+	upsertMany: (files: File[]) => void;
+	setAll: (files: File[]) => void;
+	selectAll: (filter?: Partial<File>) => File[];
+	selectById: (hashDigest: string) => File | null;
+	selectFirst: (filter: Partial<File>) => File | null;
+};
+
+function shallowMatch(
```

Comment:
do you have anything against using Ramda for transformations? It is the best functional library in js I know